### PR TITLE
Update Kubeflow Installation Script and Installation Config

### PR DESCRIPTION
**Pull request by jsitu777**
**Which issue is resolved by this Pull Request:**
* addressed comments in https://github.com/awslabs/kubeflow-manifests/pull/361; mainly update installation_config dict structure for all deployment options
* minor update in rds-s3 and cognito auto-script for new installation_config dict structure
* minor website doc fix in rds-s3 section

**Description of your changes:**

**Testing:**
- [ ] Unit tests pass
- [x] e2e tests pass
- Details about new tests (If this PR adds a new feature)
- Details about any manual tests performed

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: tests/e2e/utils/kubeflow_uninstallation.py [hunk 1]
```diff
@@ -6,49 +6,52 @@
...
-    "kubeflow-roles",
-    "kubeflow-istio-resources",
-    "kubeflow-namespace",
-    "cluster-local-gateway",
-    "oidc-authservice",
-    "dex",
-    "istio-1-14",
-    "cert-manager",
-    "aws-telemetry",
+    #"aws-authservice",
```
Comment (ryansteakley) [inline tests/e2e/utils/kubeflow_uninstallation.py:25]: why are all these commented out?

File: tests/e2e/utils/kubeflow_installation.py [hunk 2]
```diff
@@ -55,167 +56,184 @@
...
-                for kustomize_path in path_dic[component_name]["installation_options"]["kustomize"]:
-                    apply_kustomize(kustomize_path)
-            elif component_name == "ack-sagemaker-controller":
-                build_ack_controller()
-            else:
-                install_helm(component_name, installation_path, namespace)
-        # kustomize
+        print(f"==========Installing {component_name}==========")
+        # remote repo
+        if "repo"in installation_config[component_name]["installation_options"][installation_option]:
```
Comment (surajkota) [inline tests/e2e/utils/kubeflow_installation.py:109]: +1 clean

File: tests/e2e/utils/kubeflow_installation.py [hunk 3]
```diff
@@ -55,167 +56,184 @@
...
+                if component_name == "kubeflow-namespace":
+                    for kustomize_path in installation_config[component_name]["installation_options"]["kustomize"]["paths"]:
+                        apply_kustomize(kustomize_path)
+                else:
+                    install_helm(component_name, installation_paths)
+            # kustomize
+            else:
+                # crd required to established for installation
+                if "validations" in installation_config[component_name] and "crds" in installation_config[component_name]["validations"]:
+                    print("need to wait for crds....")
```
Comment (surajkota) [inline tests/e2e/utils/kubeflow_installation.py:126]: nit: remove, looks like debug log

File: tests/e2e/utils/kubeflow_installation.py [hunk 4]
```diff
@@ -55,167 +56,184 @@
...
-    retcode = subprocess.call(
-        f"helm repo add eks https://aws.github.io/eks-charts".split()
-    )
-    assert retcode == 0
-    retcode = subprocess.call(f"helm repo update".split())
-    assert retcode == 0
-    cmd = f"helm upgrade --install aws-load-balancer-controller eks/aws-load-balancer-controller \
+def install_alb_controller(cluster_name):
+    exec_shell(f"helm repo add eks https://aws.github.io/eks-charts".split())
+
```
Comment (surajkota) [inline tests/e2e/utils/kubeflow_installation.py:183]: extra spaces

File: tests/e2e/utils/kubeflow_uninstallation.py [hunk 5]
```diff
@@ -83,64 +86,74 @@ def uninstall_kubeflow(installation_option, deployment_option):
...
+                        delete_kustomize(kustomize_path)
+
+                elif component_name == "ingress":
+                    uninstall_helm(component_name, namespace)
+                    #Helm doesn't seem to delete ingress during uninstall
+                    exec_shell(f"kubectl delete ingress -n istio-system istio-ingress")
+                else:
+                    if check_helm_chart_exists(component_name, namespace):
+                        uninstall_helm(component_name, namespace)
+                    print(installation_path)
```
Comment (surajkota) [inline tests/e2e/utils/kubeflow_uninstallation.py:120]: nit: remove debug log

File: tests/e2e/utils/kubeflow_uninstallation.py [hunk 6]
```diff
@@ -83,64 +86,74 @@ def uninstall_kubeflow(installation_option, deployment_option):
...
+                        print(f"deleting {component_name} crds ...")
+                        kubectl_delete(f"{installation_path}/crds")
+                    # delete aws-load-balancer-controller crds for official helm chart
+                    if component_name == "aws-load-balancer-controller":
+                        kubectl_delete(
+                            "https://raw.githubusercontent.com/kubernetes-sigs/aws-load-balancer-controller/6d3e976e3f60dc4588c01bad036d77c127a68e71/helm/aws-load-balancer-controller/crds/crds.yaml"
+                        )
+            # kustomize
+            else:
+                installation_path.reverse()
```
Comment (surajkota) [inline tests/e2e/utils/kubeflow_uninstallation.py:131]: what is this doing?

File: tests/e2e/utils/kubeflow_uninstallation.py [hunk 7]
```diff
@@ -83,64 +86,74 @@ def uninstall_kubeflow(installation_option, deployment_option):
...
+            kubectl_delete_crd("devicetokens.dex.coreos.com")
+            kubectl_delete_crd("oauth2clients.dex.coreos.com")
+            kubectl_delete_crd("offlinesessionses.dex.coreos.com")
+            kubectl_delete_crd("passwords.dex.coreos.com")
+            kubectl_delete_crd("refreshtokens.dex.coreos.com")
+            kubectl_delete_crd("signingkeies.dex.coreos.com")
+
+        print(f"All {component_name} resources are cleared!")
+
+def uninstall_remote_component(component_name, namespace):
```
Comment (surajkota) [inline tests/e2e/utils/kubeflow_uninstallation.py:149]: unused method?

File: tests/e2e/utils/kubeflow_installation.py [hunk 8]
```diff
@@ -254,7 +279,10 @@ def build_ack_controller():
...
 
     args, _ = parser.parse_known_args()
-    
+
     install_kubeflow(
-        args.installation_option,  args.deployment_option, args.cluster_name, args.aws_telemetry
+        args.installation_option,
+        args.deployment_option,
+        args.cluster_name,
+        args.aws_telemetry,
```
Comment (surajkota) [inline tests/e2e/utils/kubeflow_installation.py:287]: can remove aws_telemetry from arg parser

**Review by surajkota — APPROVED** [review]
approving and merging with comments to test

**Pull request merged**